src/anndata/_io/specs/methods.py

```diff
@@ -498,14 +498,16 @@ def write_sparse_compressed(
 ):
     g = f.require_group(key)
     g.attrs["shape"] = value.shape
+    indptr_dtype = dataset_kwargs.get("indptr_dtype", value.indptr.dtype)
+    dataset_kwargs = {k: v for k, v in dataset_kwargs.items() if k != "indptr_dtype"}
```

Comment:
You can simplify this like so:

```suggestion
    dataset_kwargs = dict(dataset_kwargs)
    indptr_dtype = dataset_kwargs.pop("indptr_dtype", value.indptr.dtype)
```